pass/lean/scripts/const_parity.py

```
@@ -0,0 +1,77 @@
+    # 2. nodeCerts: nid -> deps
+    deps = {}
+    for m in re.finditer(r"\{ nid := (\d+), op := ([^,]+), width := (\d+), deps := \[([^\]]*)\] \}", txt):
+        deps[int(m.group(1))] = (m.group(2).strip(), [int(x) for x in re.findall(r"\d+", m.group(4))])
+
+    # 3. fv bodies: def <Top>_fv<nid> ... :=  <body lines until next def/blank>
+    bodies = {}
+    cur = None
+    for ln in lines:
+        m = re.match(r"^def \w+_fv(\d+) ", ln)
+        if m:
+            cur = int(m.group(1)); bodies[cur] = ""
```

Ruff (0.16.0)

[error] Multiple statements on one line (semicolon)

(E702)